### ISSUE TITLE
My manual social graph workaround (hopefully you can implement this)

### ISSUE THREAD

**Issue by raymondpirouz** [gnotaras/wordpress-add-meta-tags#41]
Hi there,

Currently, if I tell the plugin to automatically create full meta for social graph (Facebook/Linkedin/Twitter) the results don't work consistently across all 3 social networks so I created my own template code that I manually populate each time I create a post and when I share I get a LARGE Twitter card image on Twitter and a nice thumbnail on Linkedin AND large image on Facebook...I hope you can implement this code output in a future release:

Notes: ALL CAPS are fields that must be replaced by the plug-in. The IMAGE URL and WIDTH/HEIGHT refer to the featured image and I usually upload something that is 1200px wide and around 520px high and it works out great across all 3 social networks.

`<!-- Open Graph Meta -->`
`<meta property="og:site_name" content="WEBSITE NAME" />`
`<meta property="og:title" content="POST TITLE" />`
`<meta property="og:type" content="article" />`
`<meta property="og:url" content="URL" />`
`<meta property="og:image" content="IMAGE URL" />`
`<meta property="og:image:width" content="WIDTH" />`
`<meta property="og:image:height" content="HEIGHT" />`
`<meta property="og:description" content="DESCRIPTION." />`
`<!-- Twitter Card Meta -->`
`<meta name="twitter:card" content="summary_large_image" />`
`<meta name="twitter:site" content="@TWITTER HANDLE" />`
`<meta name="twitter:creator" content="@TWITTER HANDLE" />`
`<meta name="twitter:title" content="TITLE" />`
`<meta name="twitter:description" content="DESCRIPTION." />`
`<meta name="twitter:image" content="IMAGE URL" />`

**Comment by gnotaras** [comment]
Hello Raymond and thanks for your feedback and for contributing the metadata template.

I'm always willing to improve the plugin, but in this case I think I'm going to need some more information about the inconsistencies across the social networks. Do these involve the image sizes only or is there a problem with other parts of the metadata like the description and title?

Regarding image sizes, there has been some controversy in the past about what image sizes the plugin should use by default in the metadata. Wordpress has three default image sizes (`thumbnail`, `medium`, `large`), whose sizes are configured in _Settings->Media_, and there is also the `full` image size, which refers to the original image as uploaded by the user. Moreover, themes and plugins define their own image sizes.

By convention, Add-Meta-Tags, uses the `full` image size which corresponds to the unaltered image as uploaded by the user.

In case this is not the desired image size, it can be customized by adding the following code snippet in the theme's `functions.php`:

```php
function amt_use_custom_image_size_in_metadata( $size ) {
    return 'myimagesize';
}
add_filter( 'amt_image_size_index', 'amt_use_custom_image_size_in_metadata', 10, 1 );
add_filter( 'amt_image_size_content', 'amt_use_custom_image_size_in_metadata', 10, 1 );
add_filter( 'amt_image_size_attachment', 'amt_use_custom_image_size_in_metadata', 10, 1 );
add_filter( 'amt_image_size_product', 'amt_use_custom_image_size_in_metadata', 10, 1 );
```

The above snippet instructs Add-Meta-Tags to use the custom `myimagesize` image size.

Adding custom image sizes in WordPress is as easy as adding the following inside the theme's `MYTHEME_setup()` function:

```php
add_image_size( 'myimagesize', 220, 180 );
```

Please see more information [here](https://developer.wordpress.org/reference/functions/add_image_size/).

Regarding Twitter Cards, I see that the default card type in your template is `summary_large_image`. By default, Add-Meta-Tags generates a `summary` card. In order to generate a `summary_large_image` card by default, the relevant setting has to be enabled in the _Settings -> Metadata -> Twitter Cards Metadata_.

Moreover, please read more information about the [order in which AMT searches for images](http://www.codetrax.org/projects/wp-add-meta-tags/wiki/Metadata_Strategy#Images).

About any inconsistency in other parts of the metadata, please feel free to post more feedback and details. Also, if there are any warnings/errors by online validators, posting them here would be very helpful, but as far as I can tell the generated metadata is within the specs.

Kind Regards,
George

**Comment by raymondpirouz** [comment]
Thanks for the detailed reply George, yes originally I was uploading main images smaller than 1000px and the auto generated content on FB/LinkedIn/Twitter were inconsistent so I thought there was an issue w/the way the plugin was generating the code so I inserted my own AND also began uploading larger main images (1200px width) and everything started to work so I assumed it was the code and never went back to check if the image was the issue.

So, I removed the code from "full meta tags" box in each post and everything works as expected.....false alarm.  :)  Thanks again for taking the time to reply.
